Doc/c-api/unicode.rst

```diff
@@ -1396,6 +1396,18 @@ They all return ``NULL`` or ``-1`` if an exception occurs.
    :c:func:`PyErr_Occurred` to check for errors.
 
 
+.. c:function:: int PyUnicode_EqualToUTF8(PyObject *unicode, const char *string)
+
+   Compare a Unicode object with a UTF-8 or ASCII encoded C string
+   and return true (``1``) if they are equal, or false (``0``) otherwise.
+   If the Unicode object contains null or surrogate characters or
+   the C string is not encoded to UTF-8 or ASCII, return false (``0``) .
```

Comment:
```suggestion
   the C string is not valid UTF-8, false (``0``) is returned.
```

Reply:
I asked Serhiy to write "UTF-8 or ASCII", since I expect that most developers don't know that UTF-8 is a superset of ASCII. Also, because Python C API has a complicated set of functions with strange rules about encodings: see previous comments on this PR.

Reply:
> I asked Serhiy to write "UTF-8 or ASCII", since I expect that most developers don't know that UTF-8 is a superset of ASCII.

Hmm... why not, but someone _not_ knowing that UTF8 is a superset of ASCII should perhaps not deal with encodings at all :-)

(also, how often are you sure that a string is pure ASCII these days? even Python identifiers are now Unicode)

Reply:
> (also, how often are you sure that a string is pure ASCII these days? even Python identifiers are now Unicode)

Basically all Python identifiers are ASCII in the Python code base.

Reply:
Yeah, so what?